src/Mir/Intrinsics.hs

```diff
@@ -412,7 +412,7 @@ type LabelMap s = Map.Map BasicBlockInfo (Label s)
 
 type AdtMap = Map.Map Text.Text [Variant]
 
-data TraitMap = forall s. TraitMap (Map.Map Text.Text {- ^ trait name-} (Some (TraitImpls s)))
+data TraitMap s = TraitMap (Map.Map Text.Text {- ^ trait name-} (Some (TraitImpls s)))
```

Comment:
We need the parametrization on `s` because otherwise we won't be able to use the `Expr`s that represent VTables in the translated code.